x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2035,6 +2035,23 @@ export class RulesClient {
       } catch (e) {
         throw e;
       }
+    }
+
```

Comment:
This update to the enable logic takes into account the fact that there may be already disabled rules with no corresponding task.

If the rule is disabled with no corresponding task, it's scheduledTaskId will be undefined. In this case, we want to schedule a task on enable.

If a rule somehow has a scheduledTaskId defined but it doesn't actually exist, we want to schedule a task on enable.

Finally, if a rule has a scheduledTaskId defined and the task exists, we enable the task.